src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -0,0 +1,167 @@
+                this.enabledHentai = true;
+                break;
+            default:
+                this.enabled = false;
+                this.enabledHentai = false;
+        }
+
+        if (this.enabled) {
+            this.statusNew = true;
+            this.startDiscordPresence();
+        } else {
+            this.stopDiscordPresence();
+        }
+    }
+
+    _onSelectConnector(e) {
+        // Hentai check
+        if(e.detail.tags.includes('hentai') || e.detail.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Browsing ' + e.detail.label;
+        if (this.status.state) delete this.status.state;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onMangaClicked(e) {
+        // Hentai check
+        if(e.detail.connector.tags.includes('hentai') || e.detail.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Browsing ' + e.detail.connector.label;
+        this.status['state'] = 'Looking at ' + e.detail.title;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    _onChapterGetPages(e) {
+        // Hentai check
+        if(e.detail.manga.connector.tags.includes('hentai') || e.detail.manga.connector.tags.includes('porn')) {
+            this.hentai = true;
+        } else {
+            this.hentai = false;
+        }
+
+        this.status['details'] = 'Reading ' + e.detail.manga.title;
+        this.status['state'] = e.detail.title;
+        this.status.startTimestamp = + new Date();
+        this.statusNew = true;
+        if (this.enabled && !this.rpc) this.startDiscordPresence();
+    }
+
+    async updateStatus() {
+        if(this.rpc) {
+            if (this.enabled && this.statusNew) {
+                this.IpcBytes = this.rpc.transport.socket.bytesWritten;
+                if( !this.hentai || (this.hentai && this.enabledHentai)) {
+                    this.rpc.setActivity(this.status);
+                } else {
+                    this.statusNew = false;
+                }
+            }
+
+            // Test if IPC is still active
+            if ( this.statusNew && this.rpc.transport.socket.bytesWritten > this.IpcBytes) {
+                this.IpcBytes = this.rpc.transport.socket.bytesWritten;
+                this.statusNew = false;
+            } else if (this.rpc.transport.socket.bytesWritten == this.IpcBytes && this.statusNew) {
+                console.warn('WARNING: DiscordPresence - Lost connection to Discord.');
+                this.stopDiscordPresence();
+            }
+        }
+    }
+
+    stopDiscordPresence() {
+        this.statusNew = false;
```

Comment:
Should also `clearInterval(reference)` that otherwise continues to tick every 15 seconds